### PR TITLE
feat: update BottomSheet to be scroll enabled depending on content size

### PR DESCRIPTION
**Pull request by kathaypacific**
### Description

As the description

### Test plan

There should be no visual change for the existing components

https://github.com/valora-inc/wallet/assets/20150449/800fb29d-5779-425d-96e1-d699dd270162

### Related issues

- Fixes RET-698

### Backwards compatibility

Y

### CODE REVIEW

File: __mocks__/@gorhom/bottom-sheet.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { ScrollView } from 'react-native'
+
+module.exports = {
+  __esModule: true,
+  ...(jest.requireActual('@gorhom/bottom-sheet') as any),
+  BottomSheetScrollView: ScrollView,
```
Comment (kathaypacific) [inline __mocks__/@gorhom/bottom-sheet.ts:6]: the BottomSheetScrollView has an internal utility called `getRefNativeTag` that "grab the native tag from the native host component. It only works when the ref is pointing to a native Host component" which is a real pain for unit tests. mocking it to be a RN scrollview seems harmless enough

**Review by dievazqu — APPROVED** [review]

File: src/components/BottomSheet.tsx [hunk 2]
```diff
@@ -14,16 +14,13 @@ interface Props {
   forwardedRef: React.RefObject<GorhomBottomSheet>
   title: string
   description?: string | null
-  children?: React.ReactNode
+  children?: React.ReactNode | React.ReactNode[]
```
Comment (jeanregisser) [inline src/components/BottomSheet.tsx:17]: nit: why is this needed?

**Review by jeanregisser — APPROVED** [review]
LGTM 👍
